i < json_array_size(o) is an signed vs unsigned comparison, fix that

diff -rupN flux-accounting-0.46.0.orig/src/plugins/jj.cpp flux-accounting-0.46.0/src/plugins/jj.cpp
--- flux-accounting-0.46.0.orig/src/plugins/jj.cpp	2025-06-03 18:39:31.000000000 +0200
+++ flux-accounting-0.46.0/src/plugins/jj.cpp	2025-06-25 21:17:25.524706970 +0200
@@ -66,7 +66,7 @@ static int jj_read_vertex (json_t *o, in
 
 static int jj_read_level (json_t *o, int level, struct jj_counts *jj, int nodefactor)
 {
-    int i;
+    size_t i;
     json_t *v = NULL;
 
     if (!json_is_array (o)) {
